### PR TITLE
Improve docs for Elasticsearch 8

### PR DESCRIPTION
**Pull request by philipp94831**
The docs for connecting to Elasticsearch container only work for Elasticsearch v7. For Elasticsearch v8 either the SSLContext has to be configured for the HTTPClient or SSL needs to be disabled. This PR adds according documentation

### CODE REVIEW

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 1]
```diff
@@ -213,6 +213,85 @@ public void restClientClusterHealth() throws IOException {
...
+                        httpClientBuilder.setSSLContext(container.createSslContextFromCa());
+                        return httpClientBuilder;
+                    })
+                    .build();
+
+            Response response = client.performRequest(new Request("GET", "/_cluster/health"));
+            // }}
+            assertThat(response.getStatusLine().getStatusCode()).isEqualTo(200);
+            assertThat(EntityUtils.toString(response.getEntity())).contains("cluster_name");
+            // httpClientContainer8 {{
```
Comment (eddumelendez) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:251]:
```suggestion
            // httpClientContainer8 {
```

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 2]
```diff
@@ -213,6 +213,85 @@ public void restClientClusterHealth() throws IOException {
...
+            client =
+                RestClient
+                    .builder(HttpHost.create(container.getHttpHostAddress()))
+                    .setHttpClientConfigCallback(httpClientBuilder -> {
+                        return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                    })
+                    .build();
+
+            Response response = client.performRequest(new Request("GET", "/_cluster/health"));
+            // }}
```
Comment (eddumelendez) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:287]:
```suggestion
            // }
```

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 3]
```diff
@@ -213,6 +213,85 @@ public void restClientClusterHealth() throws IOException {
...
+            assertThat(response.getStatusLine().getStatusCode()).isEqualTo(200);
+            assertThat(EntityUtils.toString(response.getEntity())).contains("cluster_name");
+            // httpClientContainer8NoSSL {{
```
Comment (eddumelendez) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:290]:
```suggestion
            // httpClientContainer8NoSSL {
```

**Comment by eddumelendez** [comment]
Hi @philipp94831, thanks for your contribution. I have added some suggestions that will fix how the code is render in the documentation, see https://deploy-preview-8870--testcontainers.netlify.app/modules/elasticsearch/

Also, regarding to the new tests I think `restClientClusterHealthElasticsearch8` is the same as 
`testElasticsearch8SecureByDefault` but it is reusing a method, which doesn't make a good candidate to be exposed in the docs. However, let's avoid duplication.

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java (hunk 1 above)
Comment (philipp94831) [reply modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:251]: In line 211 it is also `{{`. Which one is correct?

**Comment by philipp94831** [comment]
> Hi @philipp94831, thanks for your contribution. I have added some suggestions that will fix how the code is render in the documentation, see [deploy-preview-8870--testcontainers.netlify.app/modules/elasticsearch](https://deploy-preview-8870--testcontainers.netlify.app/modules/elasticsearch/)
> 
> Also, regarding to the new tests I think `restClientClusterHealthElasticsearch8` is the same as `testElasticsearch8SecureByDefault` but it is reusing a method, which doesn't make a good candidate to be exposed in the docs. However, let's avoid duplication.

What is your suggestion wrt duplication? I mainly added the tests in order to document the connection to ES 8. Because of the method used in the existing tests I deemed them unsuitable for documentation purposes

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java (hunk 1 above)
Comment (kiview) [reply modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:251]: It should be a single curly brace.

**Comment by kiview** [comment]
I think if we want to have `restClientClusterHealthElasticsearch8` for docs purposes, we can remove `testElasticsearch8SecureByDefault`, WDYT @eddumelendez?

**Comment by eddumelendez** [comment]
Sorry for the delay. Let's proceed with @kiview suggestions. Also, you can check docs locally by running `docker compose up` or when submitting the PR, check `netlify/testcontainers/deploy-preview`

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java (hunk 1 above)
Comment (philipp94831) [reply modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:251]: I tested it and it only works with a single curly brace here. Docs preview looks good to me now

**Review by eddumelendez — DISMISSED** [review]

**Comment by eddumelendez** [comment]
@philipp94831 can you please run `./gradlew spotlessApply`?

**Comment by eddumelendez** [comment]
Thanks @philipp94831 !

**Pull request merged**